src/main/scala/mesosphere/marathon/tasks/ResourceUtil.scala

```diff
@@ -1,121 +1,151 @@
 package mesosphere.marathon.tasks
 
 import com.twitter.util.NonFatal
+import mesosphere.marathon.state.DiskSource
+import mesosphere.mesos.protos
+import org.apache.mesos.Protos.Resource.DiskInfo.Source
 import org.apache.mesos.Protos.Resource.{ DiskInfo, ReservationInfo }
 import org.apache.mesos.{ Protos => MesosProtos }
 import org.slf4j.LoggerFactory
 import scala.collection.JavaConverters._
 
 object ResourceUtil {
+  implicit class RichResource(resource: MesosProtos.Resource) {
+    def getDiskSourceOption: Option[Source] =
+      if (resource.hasDisk && resource.getDisk.hasSource)
+        Some(resource.getDisk.getSource)
+      else
+        None
+
+    def getStringification: String = {
```

Comment:
well... 2 reasons:

- it's an implicit class; you can't provide new implementations of existing methods with an implicit class
- it's a special type of stringification. That's what the mesos internal code calls it when it takes a string and prevents it in string form. I've got a pending merge request (that I need to tend to :( and am neglecting :( :( ) to have mesos output this same format for the multi disk resources.